src/__tests__/github_issues/231-test.js

```diff
@@ -0,0 +1,21 @@
+/* @flow */
+
+import { schemaComposer } from '../..';
+
+describe('github issue #231: Cloning a resolver evaluates its configuration thunks', () => {
+  it('clone a resolver without evaluating its type and args thunks', async () => {
+    const aTypeThunk = jest.fn(() => 'String');
+
+    const aResolver = schemaComposer.createResolver({
+      name: aTypeThunk,
+      type: 'Boolean',
+      args: {
+        anArg: aTypeThunk,
+      },
+    });
```

Comment:
There is a problem!

When you provide argument `anArg: aTypeThunk` internally `graphql-compose` must construct argument config. BUT was provided a function. So it does not know what is a result of function – `ready argument config` OR `just some type`. So it starts evaluating it.

But if you provide explicitly `anArg: { type: aTypeThunk }` it exactly knows that you're providing a type, so no need in function resolving.

Anyway i think that it's quite strange to have ability to provide argument config as a thunk.

Reply:
> When you provide argument anArg: aTypeThunk internally graphql-compose must construct argument config.

But if I just create a resolver without using it, they're not evaluated (that's useful when constructing the graph with circular dependencies). So why .clone() should evaluate those?
After all, the clone function I use as a workaround does not have this problem and still provides a working resolver in the end. But I don't know the internals enough. Maybe you're doing a deep copy when cloning, whereas for me, just reassigning the `Resolver.args` property (ie. not using `Resolver.setArgs` which causes the problem) is enough.

> But if you provide explicitly anArg: { type: aTypeThunk } it exactly knows that you're providing a type, so no need in function resolving.

Yes indeed, that works, thanks!

> Anyway i think that it's quite strange to have ability to provide argument config as a thunk.

Well, at least for me that was an useful shorthand... having to specify an object with `{ type: () => /**/ }` adds visual clutter and verbosity.

But if that's the only proper way to do that, well that's no problem.

Reply:
I've removed ArgConfig as a thunk & FieldConfig as thunk support due to getting rid of this ambiguity.